app/src/main/java/com/google/android/sambadocumentsprovider/ShareManager.java

```diff
@@ -226,48 +280,51 @@ private static String decode(String content, Map<String, CredentialTuple> creden
     }
   }
 
-  private static CredentialTuple decodeTuple(JsonReader reader) throws IOException {
+  private static ShareTuple decodeTuple(JsonReader reader) throws IOException {
     if (reader.peek() == JsonToken.NULL) {
       reader.nextNull();
-      return CredentialTuple.EMPTY_TUPLE;
+      return ShareTuple.EMPTY_TUPLE;
     }
 
     String workgroup = null;
     String username = null;
     String password = null;
+    boolean mounted = true;
 
     reader.beginObject();
     while (reader.hasNext()) {
       String name = reader.nextName();
-      String value = reader.nextString();
 
       switch (name) {
         case WORKGROUP_KEY:
-          workgroup = value;
+          workgroup = reader.nextString();
           break;
         case USERNAME_KEY:
-          username = value;
+          username = reader.nextString();
           break;
         case PASSWORD_KEY:
-          password = value;
+          password = reader.nextString();
           break;
+        case MOUNT_KEY:
+          mounted = reader.nextBoolean();
         default:
           Log.w(TAG, "Ignoring unknown key " + name);
       }
     }
     reader.endObject();
 
-    return new CredentialTuple(workgroup, username, password);
+    return new ShareTuple(workgroup, username, password, mounted);
   }
 
-  private static class CredentialTuple {
-    private static final CredentialTuple EMPTY_TUPLE = new CredentialTuple("", "", "");
+  private static class ShareTuple {
+    private static final ShareTuple EMPTY_TUPLE = new ShareTuple("", "", "", false);
```

Comment:
mounted value for EMPTY_TUPLE should be `true` right? To keep existing public shares mounted.

Maybe we should get rid of EMPTY_TUPLE 'cause we always have mounted info. Thoughts?